app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt

```diff
@@ -210,10 +245,20 @@ class CreateRecipeViewModel(
    *
    * @return The instructions for the recipe.
    */
-  fun getRecipeInstructions(): String {
+  fun getRecipeInstructions(): List<Instruction> {
```

Comment:
If this doesnt take too much time, is it possible to refactor the function name ? it looks very similar to the `getRecipeInstruction(index: Int): Instruction` one. I'd advise you to deal with the other points I raised in this review before doing this one, which I don't think is the most important yet.

Reply:
done